airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -704,7 +728,10 @@ def request_params(
     ) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
         if not next_page_token:
-            params.update({"createdBefore": pendulum.now("utc").strftime(self.time_format)})
+            end_date = pendulum.now("utc").strftime(self.time_format)
+            if self._replication_end_date:
+                end_date = self._replication_end_date
+            params.update({self.replication_end_date_field: end_date})
```

Comment:
On this stream the `replication_start_date_field` might be automatically set compared to `now` because *The date range to search must not be more than 7 days.*
I think you might also need to change the computation of the `replication_start_date_field` in the `__init__` if `_replication_end_date` is set.

Reply:
After giving a closer look, I doubt that this API ever worked on Airbyte.
We are not setting `createdAfter` request parameter, which is `required`

```
https://sellingpartnerapi-na.amazon.com/vendor/directFulfillment/shipping/v1/shippingLabels?createdBefore=2022-05-24T10%3A26%3A22Z
```

`self.replication_start_date_field = max(...)`
This code is overwriting the value `createdAfter`, which should have been used to set request params. 

We should move the logic to calculate `start date` to `request_params(...)` along with calculation and setting `end date`


Please let me know if my observation is correct and will make the necessary changes.

Reply:
I have made the changes. Please review and share your thoughts.